server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -579,6 +581,15 @@ public Query existsQuery(QueryShardContext context) {
             }
         }
 
+        @Override
+        public IntervalsSource intervals(String text, int maxGaps, boolean ordered, NamedAnalyzer analyzer) throws IOException {
+            if (indexOptions().compareTo(IndexOptions.DOCS_AND_FREQS_AND_POSITIONS) < 0) {
+                throw new IllegalArgumentException("Cannot create source against field [" + name() + "] with no positions indexed");
```

Comment:
maybe say "intervals source" instead of just source?

Reply:
++